src/Symfony/Component/HttpFoundation/Session/Storage/Handler/NativeFileSessionHandler.php

```diff
@@ -48,8 +48,8 @@ public function __construct($savePath = null)
             $baseDir = ltrim(strrchr($savePath, ';'), ';');
         }
 
-        if ($baseDir && !is_dir($baseDir)) {
-            mkdir($baseDir, 0777, true);
+        if ($baseDir && !is_dir($baseDir) && !@mkdir($baseDir, 0777, true) && !is_dir($baseDir)) {
+            throw new \RuntimeException(sprintf('The Session Storage was not able to create a directory: %s', $baseDir));
```

Comment:
did anyone report the need for these exceptions? I'm sure something already fails when the dir can't be created. Did you experience any "unclear" error?